### DISCUSSION TITLE
Allow for multiple currency payment of one invoice

### DISCUSSION THREAD

**Discussion by ndeet** [btcpayserver/btcpayserver#2826] [Ideas / Feature Requests]
A use case that comes up with using Liquid Assets as promotional or pre-paid tokens:

Your customer bought a pre-sale item and got a Liquid Asset token (voucher) for it with which the item can be payed later when in stock. This works fine for digital products but for physical products the problem is that you as store owner still need/want to collect shipping fees which vary a lot for international shipping. 

### Problem: 
You get one token for a product Z with price X. A cart could look like this:
- Product Z .... 20 $
- Shipping ...... 10 $

So the invoice would be 30 $

If you allow multiple currencies the one voucher token would only reduce the amount by 1 or even not that as those tokens do not have any exchange rates to $.

### Possible solution:
Ideally you should be able to create an invoice where you can set voucher token to amount X and the expected amount of those tokens. Above example: expect 1 voucher token accounts internally for 20$, the rest is $ and can be paid by currencies with exchange rates (or a specific one set via API).

Likely this will result in major refactoring of the payment flow, API etc. but it would be great to get your opinion on it or an alternative solution.

**Comment by Kukks** [comment]
Set a rate for productZ to $ in store settings=>Rates=>Advanced rate scripting=> 
Add
```
PRODUCTZ_USD = 20;
PRODUCTZ_X = PRODUCTZ_USD * USD_X;
```
